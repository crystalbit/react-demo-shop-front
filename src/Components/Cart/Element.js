import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import Typography from '@material-ui/core/Typography';
import AddIcon from '@material-ui/icons/Add';
import RemoveIcon from '@material-ui/icons/Remove';
import config from '../../config';
import path from 'path';

const useStyles = makeStyles(theme => ({
  root: {
      flexGrow: 1,
      marginBottom: 5,
      marginLeft: 8
    },
    paper: {
      padding: theme.spacing(2),
      margin: 'auto'
    },
    image: {
      width: 128,
      height: 90,
    },
    img: {
      margin: 'auto',
      display: 'block',
      maxWidth: '100%',
      maxHeight: '100%',
    }
}));

export default function(props) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Paper className={classes.paper}>
        <Grid container spacing={2}>
          <Grid item>
            <div className={classes.image}>
              <img className={classes.img} alt={props.name} src={path.join(config.directories.images, props.image)} />
            </div>
            <div>
              <ButtonGroup className={classes.btngroup}>
                <Button
                  aria-label="reduce"
                  onClick={() => props.onLess(props.id)}
                >
                  <RemoveIcon fontSize="small" />
                </Button>
                <Button
                  aria-label="increase"
                  onClick={() => props.onMore(props.id)}
                >
                  <AddIcon fontSize="small" />
                </Button>
              </ButtonGroup>
            </div>
          </Grid>
          <Grid item xs={12} sm container>
            <Grid item xs container direction="column" spacing={2}>
              <Grid item xs>
                <Typography gutterBottom variant="subtitle1">
                  {props.name}
                </Typography>
                <Typography variant="body2" gutterBottom>
                  {props.quantity} x ${props.price} = ${(props.quantity * props.price).toFixed(2)}
                </Typography>
              </Grid>
              <Grid item>
                <Button variant="outlined" onClick={() => props.onDelete(props.id)}>
                  Remove
                </Button>
              </Grid>
            </Grid>
            <Grid item>
              <Typography variant="subtitle1">
                ${(props.quantity * props.price).toFixed(2)}
              </Typography>
            </Grid>
          </Grid>
        </Grid>
      </Paper>
    </div>
  );
}
