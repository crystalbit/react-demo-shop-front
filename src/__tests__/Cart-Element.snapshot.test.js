import React from 'react';
import renderer from 'react-test-renderer';
import Element from '../Components/Cart/Element';

test('Cart Element equals its snaphot', () => {
  const component = renderer.create(
    <Element
      id={1}
      name="A Product"
      onLess={console.log}
      onMore={console.log}
      onDelete={console.log}
      quantity={1}
      price={10}
      image="california.jpg"
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
