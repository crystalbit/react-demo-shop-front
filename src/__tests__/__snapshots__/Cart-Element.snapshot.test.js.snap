// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Element equals its snaphot 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiPaper-root makeStyles-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="makeStyles-image-3"
        >
          <img
            alt="A Product"
            className="makeStyles-img-4"
            src="/uploads/california.jpg"
          />
        </div>
        <div>
          <div
            className="MuiButtonGroup-root"
            role="group"
          >
            <button
              aria-label="reduce"
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13H5v-2h14v2z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="increase"
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlined"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true"
        >
          <div
            className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-gutterBottom"
            >
              A Product
            </h6>
            <p
              className="MuiTypography-root MuiTypography-body2 MuiTypography-gutterBottom"
            >
              1
               x $
              10
               = $
              10.00
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Remove
              </span>
            </button>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item"
        >
          <h6
            className="MuiTypography-root MuiTypography-subtitle1"
          >
            $
            10.00
          </h6>
        </div>
      </div>
    </div>
  </div>
</div>
`;
